consensus/core/src/network/metrics.rs

```diff
@@ -0,0 +1,412 @@
+// better resolution at high sizes.
+const SIZE_BYTE_BUCKETS: &[f64] = &[
+    2048., 8192., // *4
+    16384., 32768., 65536., 131072., 262144., 524288., 1048576., // *2
+    1572864., 2359256., 3538944., // *1.5
+    4600627., 5980815., 7775060., 10107578., 13139851., 17081807., 22206349., 28868253., 37528729.,
+    48787348., 63423553., // *1.3
+];
+
+impl NetworkMetrics {
+    pub fn new(direction: &'static str, registry: &Registry) -> Self {
+        let requests = register_int_counter_vec_with_registry!(
+            format!("{direction}_requests"),
+            "The number of requests made on the network",
+            &["route"],
+            registry
+        )
+        .unwrap();
+
+        let request_latency = register_histogram_vec_with_registry!(
+            format!("{direction}_request_latency"),
+            "Latency of a request by route",
+            &["route"],
+            LATENCY_SEC_BUCKETS.to_vec(),
+            registry,
+        )
+        .unwrap();
+
+        let request_size = register_histogram_vec_with_registry!(
+            format!("{direction}_request_size"),
+            "Size of a request by route",
+            &["route"],
+            SIZE_BYTE_BUCKETS.to_vec(),
+            registry,
+        )
+        .unwrap();
+
+        let response_size = register_histogram_vec_with_registry!(
+            format!("{direction}_response_size"),
+            "Size of a response by route",
+            &["route"],
+            SIZE_BYTE_BUCKETS.to_vec(),
+            registry,
+        )
+        .unwrap();
+
+        let excessive_size_requests = register_int_counter_vec_with_registry!(
+            format!("{direction}_excessive_size_requests"),
+            "The number of excessively large request messages sent",
+            &["route"],
+            registry
+        )
+        .unwrap();
+
+        let excessive_size_responses = register_int_counter_vec_with_registry!(
+            format!("{direction}_excessive_size_responses"),
+            "The number of excessively large response messages seen",
+            &["route"],
+            registry
+        )
+        .unwrap();
+
+        let inflight_requests = register_int_gauge_vec_with_registry!(
+            format!("{direction}_inflight_requests"),
+            "The number of inflight network requests",
+            &["route"],
+            registry
+        )
+        .unwrap();
+
+        let errors = register_int_counter_vec_with_registry!(
+            format!("{direction}_request_errors"),
+            "Number of errors by route",
+            &["route", "status"],
+            registry,
+        )
+        .unwrap();
+
+        Self {
+            requests,
+            request_latency,
+            request_size,
+            response_size,
+            excessive_size_requests,
+            excessive_size_responses,
+            inflight_requests,
+            errors,
+        }
+    }
+}
+
+#[derive(Clone)]
+pub struct MetricsMakeCallbackHandler {
```

Comment:
This seems anemo specific. We should have some separations between implementation specific types. For example this can be defined in anemo_manager.rs.